src/hypervisor.c

```diff
@@ -46,6 +47,228 @@
+
+        g_debug ("cor child netns path %s", netns_path);
+
+        iface_dir_path = g_strdup_printf("/sys/bus/pci/devices/%s/net", d_info->bdf);
+
+	d = opendir(iface_dir_path);
+        if (!d) {
+                g_debug ("opening net dir on bdf %s failed", d_info->bdf);
+		goto out;
+        }
+
+        while ((dir = readdir(d)) != NULL) {
+                if (g_strcmp0 (dir->d_name, ".") &&
+                        g_strcmp0 (dir->d_name, "..")) {
+                        break;
+                }
+        }
+
+        iface_name = g_strdup(dir->d_name);
+	g_debug ("iface name for bdf %s - %s", d_info->bdf, iface_name);
+
+	fork_pid = fork();
+	if (!fork_pid) {
+		/* child process */
+		char * arg[4];
+		pid_str = g_strdup_printf("%d", Pid);
+
+                arg[0] = "/usr/bin/cc-sriovdownscript.sh";
+                arg[1] = pid_str;
+                arg[2] = iface_name;
+                arg[3] = NULL;
+                if (execvp (arg[0], &arg[0]) < 0) {
+                        g_debug (" failed to execute the down script");
+                }
+                _exit(1);
+
+        } else if (fork_pid > 0) {
+		/* parent process - wait for exec to finish */
+                while (waitpid(fork_pid, &status, 0) != fork_pid);
+
+                if (WIFEXITED(status) && WEXITSTATUS(status) == 0) {
+                   g_debug("sriovdownscript existed with success");
+                }
+        }
+
+        retval = true;
+out:
+	g_free(iface_dir_path);
+	g_free(iface_name);
+	g_free(netns_path);
+	g_free(pid_str);
+	closedir(d);
+
+	return retval;
+}
+/*!
+ * Bind given device to host.
+ *
+ * \param d_info \ref cc_oci_device.
+ */
+gboolean
+cc_oci_bind_host (struct cc_oci_device* d_info)
+{
+        FILE* f;
+        gchar *bind_driver_path = NULL, *unbind_driver_path = NULL;
+	gboolean retval = false;
+
+	bind_driver_path = g_strdup_printf("/sys/bus/pci/drivers/%s/bind",
+				d_info->driver);
+	unbind_driver_path = g_strdup_printf("/sys/bus/pci/devices/%s/driver/unbind",
+				d_info->bdf);
+        g_debug ("driver path for binding %s %s", d_info->bdf, bind_driver_path);
+
+        f = fopen(unbind_driver_path, "w");
+        if (!f) {
+		g_debug ("cor: opening %s failed", unbind_driver_path);
+		g_debug ("cor: file open error %d", errno);
+		goto out;
+        }
+        fprintf(f, "%s", d_info->bdf);
+        fclose(f);
+
+        f = fopen (bind_driver_path, "w");
+        if (!f) {
+		g_debug ("cor: opening %s failed", bind_driver_path);
+		g_debug ("cor: file open error %d", errno);
+		goto out;
+        }
+        fprintf(f, "%s", d_info->bdf);
+        fclose(f);
+
+	retval = true;
+out:
+	g_free(bind_driver_path);
+	g_free(unbind_driver_path);
+        return retval;
+}
+
+/*!
+ * Unbind a given interface from the host and add it as
+ * a vfio pci device for the guest.
+ *
+ * \param config \ref cc_oci_config*
+ * \param index \ref guint
+ *
+ */
+gboolean
+cc_oci_unbind_host(struct cc_oci_config *config, guint index)
```

Comment:
Agreed- will do a third patch set to take this and some of the functional changes into account.